docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc

```diff
@@ -36,6 +36,34 @@ refer to <<ml-nlp-overview>>.
 * https://huggingface.co/roberta-large
 * https://huggingface.co/microsoft/mpnet-base
 
+[discrete]
+=== Fill mask expected model output
+
+Fill mask is a specific kind of token classification. Usually the base training task of many transformer models.
+
+For the Elastic stack's fill mask NLP task to understand the model output, it must have a specific format. 
+
+It needs to be <number of sequences inferred> by <number of tokens in sequence> by <token score for all tokens in vocabulary>.
```

Comment:
> It needs to be <number of sequences inferred> by <number of tokens in sequence> by <token score for all tokens in vocabulary>.

I think this sentence should be joined to the previous paragraph. I also think there's some words/concepts missing here. For example: "It is a nested set of {arrays? objects? tuples?} for each inferred sequence, where every token in every sequence has token scores for each token in the vocabulary.

Reply:
The reader shouldn't be reading this unless they understand machine learning model outputs and are wondering how to provide their own to conform to our pipeline. 

I am not sure we need to be super vocab friendly here.

Technically, it's nested sets of float tensors of specific dimension related to the input and configuration. I will see what other frameworks do for their output pipeline specification documentation.